File: IM-APP-server/internal/models/feedback.go
package models

type CreateFeedbackRequest struct {
	Contact      string   `json:"contact,omitempty"`
	Content      string   `json:"content"`
	ImageFileIDs []string `json:"imageFileIds,omitempty"`
}

type FeedbackResult struct {
	ID        string `json:"id"`
	CreatedAt string `json:"createdAt"`
}

File: IM-APP-fronend/src/api/feedback.ts
import { request } from '@/utils/request'

export interface CreateFeedbackInput {
  contact?: string
  content: string
  imageFileIds?: string[]
}

export interface FeedbackResult {
  id: string
  createdAt: string
}

export async function createFeedback(input: CreateFeedbackInput): Promise<FeedbackResult> {
  return request<FeedbackResult>({
    url: '/feedbacks',
    method: 'POST',
    data: {
      contact: input.contact || '',
      content: input.content,
      imageFileIds: input.imageFileIds || [],
    },
  })
}
